### PR TITLE
#7942 test_helpconfig.py migrate from testdir to pytester

### PR DESCRIPTION
**Pull request by proggga**

### CODE REVIEW

**Comment by proggga** [comment]
Okey, I need help with #7942
How I can replace testdir.monkeypath and testdir.tmpdir in the right way?

**Comment by proggga** [comment]
Hi, Ive made this replace, but idk why its failing, don't get why, trying to debug, if someone can help, it will be nice

**Review by bluetech — COMMENTED** [review]
> Hi, Ive made this replace, but idk why its failing, don't get why, trying to debug, if someone can help, it will be nice

Your changes are all correct, except one comment.

The tests failing are not your fault, turns out there are some special cases for `testdir`. I submitted #7948 to fix those; after it is merged, I will ask you to rebase this PR and then it should pass.

File: testing/test_helpconfig.py [hunk 1]
```diff
@@ -69,46 +70,46 @@ def pytest_addoption(parser):
...
-def test_traceconfig(testdir):
-    result = testdir.runpytest("--traceconfig")
+def test_traceconfig(pytester: Pytester) -> None:
+    result = pytester.runpytest("--traceconfig")
     result.stdout.fnmatch_lines(["*using*pytest*py*", "*active plugins*"])
 
 
-def test_debug(testdir):
-    result = testdir.runpytest_subprocess("--debug")
+def test_debug(pytester: Pytester, tmpdir) -> None:
```
[the remark this answers is not in the document]
Comment (proggga) [reply testing/test_helpconfig.py]: Thanks

**Comment by nicoddemus** [comment]
#7948 has been merged, can you please rebase @proggga (squashing commits)? Thanks!

**Comment by proggga** [comment]
> #7948 has been merged, can you please rebase @proggga (squashing commits)? Thanks!

Fixed all issues, rebased and squashed

**Review by bluetech — APPROVED** [review]
Thanks!

**Pull request merged**